File: 1283.find-the-smallest-divisor-given-a-threshold/solution.go
func smallestDivisor(nums []int, threshold int) int {
    return 1 + sort.Search(1000000, func(i int) bool {
        sum := 0
        for _, num := range nums {
            sum += (num + i) / (i + 1)
        }
        return sum <= threshold
    })
}

File: 1283.find-the-smallest-divisor-given-a-threshold/solution.kt
class Solution {
    fun smallestDivisor(nums: IntArray, threshold: Int): Int {
        var l = 1
        var r = 1000000
        while (l < r) {
            val divisor = l + (r - l) / 2
            val sum = nums.sumOf { (it + divisor - 1) / divisor }
            if (sum <= threshold) {
                r = divisor
            } else {
                l = divisor + 1
            }
        }
        return l
    }
}